src/http/modules/ngx_http_limit_req_module.c

```diff
@@ -1036,22 +934,47 @@ ngx_http_limit_req_zone(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
     }
 
 #if (T_LIMIT_REQ)
-    v = ngx_array_push(variables);
-    if (v == NULL) {
+    {
+    u_char      *p;
+    size_t      len;
+    ngx_str_t   v1;
+
+    len = 0;
+    for (i = 1; i < cf->args->nelts; i++) {
+        if (i == 1 || value[i].data[0] == '$') {
+            len += value[i].len;
+        }
+        break;
+    }
+
+    p = ngx_palloc(cf->pool, len);
+    if (p == NULL) {
         return NGX_CONF_ERROR;
     }
 
-    value[1].len--;
-    value[1].data++;
+    v1.len = len;
+    v1.data = p;
+
+    for (i = 1; i < cf->args->nelts; i++) {
+        if (i == 1 || value[i].data[0] == '$') {
+            p = ngx_cpymem(p, value[i].data, value[i].len);
+        }
+        break;
 
-    v->index = ngx_http_get_variable_index(cf, &value[1]);
-    if (v->index == NGX_ERROR) {
-        return NGX_CONF_ERROR;
     }
 
-    v->var = value[1];
+    ngx_memzero(&ccv, sizeof(ngx_http_compile_complex_value_t));
 
+    ccv.cf = cf;
+    ccv.value = &v1;
+    ccv.complex_value = &ctx->key;
+
+    if (ngx_http_compile_complex_value(&ccv) != NGX_OK) {
+        return NGX_CONF_ERROR;
+    }
+    }
```

Comment:
Delete original implementation of multi variables.

It combine multi variables into one complex value, and reuse nginx complex value logic.